.github/workflows/ci_test-full.yml

```
@@ -25,7 +28,7 @@ jobs:
           # nightly: add when there's a release candidate
           #- {os: ubuntu-20.04, python-version: "3.10", requires: "latest", release: "pre"}
 
-    timeout-minutes: 40
+    timeout-minutes: 50
```

The current timeout seems to be working well at the moment, the hanging MacOS test is due to a bug: #11365

Let's keep this PR focused just to the concurrency feature

```suggestion
    timeout-minutes: 40
```